pkg/controller/vsphere/machine_scope.go

```diff
@@ -130,6 +130,44 @@ func (s *machineScope) GetSession() *session.Session {
 	return s.session
 }
 
+func (s *machineScope) isNodeLinked() bool {
+	return s.machine.Status.NodeRef != nil && s.machine.Status.NodeRef.Name != ""
+}
+
+func (s *machineScope) getNode() (*apicorev1.Node, error) {
+	var node apicorev1.Node
+	if !s.isNodeLinked() {
+		return nil, fmt.Errorf("NodeRef empty, unable to get related Node")
+	}
+	nodeName := s.machine.Status.NodeRef.Name
+	objectKey := runtimeclient.ObjectKey{
+		Name: nodeName,
+	}
+	if err := s.apiReader.Get(s.Context, objectKey, &node); err != nil {
+		if apimachineryerrors.IsNotFound(err) {
+			klog.V(2).Infof("Node %q not found", nodeName)
+			return nil, err
+		}
+		klog.Errorf("Failed to get node %q: %v", nodeName, err)
+		return nil, err
+	}
+
+	return &node, nil
+}
+
+func (s *machineScope) checkNodeReachable() (bool, error) {
```

Comment:
Same concern as for other functions https://github.com/openshift/machine-api-operator/blob/f4110d923215831c8d47d846f82b5d8ce156c2e8/pkg/controller/machine/controller.go#L526